h2o-core/src/main/java/water/RestApiPingCheckThread.java

```diff
@@ -0,0 +1,33 @@
+package water;
+
+import water.api.PingHandler;
+import water.util.Log;
+
+public class RestApiPingCheckThread extends Thread {
+  
+  public RestApiPingCheckThread() {
+    this.setDaemon(true);
+  }
+
+  @Override
+  public void run() {
+    while (!Thread.currentThread().isInterrupted()) {
+      if (H2O.CLOUD._memary.length != 0 && H2O.SELF == H2O.CLOUD.leader()) {
```

Comment:
If the H2O node is not a loader, can't the thread just regularly finish?

Reply:
@michalkurka can leader node change during H2O run-time? I think it can, but I guess that after cloud is locked the leader will remain same.

If that is the case, we can close the thread after cloud is locked in case we are not leader

Reply:
the leader doesn't change after the cloud is locked

Reply:
perfect!!